src/integer/poly_over_z/from.rs

```diff
@@ -83,6 +86,79 @@ impl FromStr for PolyOverZ {
     }
 }
 
+impl PolyOverZ {
+    /// Create a [`PolyOverZ`] from a [`PolyOverZq`].
+    ///
+    /// Parameters:
+    /// - `poly`: the polynomial from which the coefficients are copied
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::PolyOverZ;
+    /// use qfall_math::integer_mod_q::PolyOverZq;
+    /// use std::str::FromStr;
+    ///
+    /// let poly = PolyOverZq::from_str("4  0 1 102 3 mod 101").unwrap();
+    ///
+    /// let poly_z = PolyOverZ::from_poly_over_zq(&poly);
+    ///
+    /// # let cmp_poly = PolyOverZ::from_str("4  0 1 1 3").unwrap();
+    /// # assert_eq!(cmp_poly, poly_z);
+    /// ```
+    pub fn from_poly_over_zq(poly: &PolyOverZq) -> Self {
```

Comment:
For the other types, we moved away from the explicit `from_xzy` functions and use only the From trait. I think that this should also be done here and for the other cases.